### ISSUE TITLE
🚀 [Feature Request] AQ War Effort

### ISSUE THREAD

**Issue by killerwife** [cmangos/issues#2209]
Progress on AQ War Effort requires the following:

Spawns and gossips for the item NPCs:
15383,15431,15432,15434,15437,15445,15446,15448,15450,15451,15452,15453,15455,15456,15457,15459,15460,15469,15477,15508,
15512,15515,15522,15525,15528,15529,15532,15533,15534,15535

they need 2 gossips, when on game_event 120 they need to have the "items" texts, when on game_event 121 they need to have the "resources are travelling to silithus in X days".

queries that can be used to find the texts in broadcast_text:
select * from broadcast_text where Text1 like '%$1%';
select * from broadcast_text where Text1 like '%$2%';

Gong spawn for this q:
https://classic.wowhead.com/quest=8743/bang-a-gong

After I have all of this data I will progress to stage 2, test it and push it to all cores. After that it will need the scripting of 10 hour war and miscellaneous cosmetic GOs for progressing the resources, but the skeleton will be done and I will put all notes here on what I know will need to be done.

@cala and @Tobschinski if you can get me this data as soon as possible that would be great.

**Comment by cala** [comment]
Sorry, this one went under my radar as I was extremely busy at the time. After looking in my tabs, it appears that unfortunately I have close to no data regarding AQ War Effort. 😢

**Comment by killerwife** [comment]
I have sniffed the event and I roughly know whats going on, and my structure for it will make it to cmangos eventually but I have no time to process the data myself. Too many tbc things keeping me occupied.

**Comment by cala** [comment]
If your sniff data are already parsed, I can look at making them into neat SQL queries to add them into our databases.

This will not be in a near future though, as I have a few issues/PR to merge first and I'm about to start working on C'Thun which will probably not be completed in a single week.

**Comment by killerwife** [comment]
Hah, so you are beating me to cthun eh :D Alright. Sniffs are parsable and I have access to what @ratkosrb has at vmangos, just no time to do it.

**Comment by killerwife** [comment]
Official TODO list for this task.
Currently, counter handling is done in core and pushed to all 3 cores. There are some game_event ids reserved for it inside game/World/WorldState.h

The rough structure of the event.

1. War effort starts. Quests are available, counters count all finished quests. As the counters progress, supplies spawn around the vendors as unclickable (I imagine) gameobjects. Notable things are two patroling NPCs in each zone, Orgrimmar and Ironforge. There are also plenty of npcs with counters through worldstates in their texts.
2. Once ALL quests are done, an exact 5 day period timer starts, during which the resources slowly diminish. Unconfirmed, some npcs show up in silithus.
3. At the end of the timer, two things happen, I am vague on the order. Either A - someone rings gong, gong event starts and then war breaks out OR B a band of npcs starts walking towards the gate and they enable the gong. After this, kalimdorwide event spawns, in silithus some big bosses spawn and a lot of elite npcs and in Barrens, Feralas, 10k needles and Tanaris huge pillars rise up with defendants which drop above average amount of rare loot and give reputation.
4. 10 hours after the war has started, gong no longer gives mount and title (only tbc+) and all the attackers despawn

Needs to work across all 3 cores, and controlled through chat commands so each stage can be reset for testing.

**Comment by killerwife** [comment]
Placed a $60 USB bounty on this issue as per me and my team. https://www.bountysource.com/issues/92474284-feature-request-aq-war-effort

**Comment by killerwife** [comment]
Clarification on closing this issue, two people need to approve this after review. Both core and DB component.

**Comment by cala** [comment]
Though I would love to see this in-game event completed as much as everyone here do, I'm uneasy with the bounty.

I've only worked on scientific patents IP, not the artistic content IP used in video games so I'm not adamant on this, but I believe that giving money (or asking for money) in exchange of work that includes copyrighted material not owned the involved parties (or licenced to them), is beyond the legal line in most countries.

One would argue that the amount of money and copyrighted material is small - and I agree with that - but if one day the copyright owner decides to run a DMCA offensive similar to the one RIAA and YouTube are currently running on open source and free projects like [youtube-dl and its forks](https://github.com/github/dmca/blob/master/2020/10/2020-10-23-RIAA.md), this bounty would be like a Christmas gift for their lawyers as we won't be able to protect ourselves by claiming CMaNGOS is a educational project, because someone is making profit from it.

CMaNGOS was already taken down once on GitHub by a DMCA request and we know that the copyright owner already closed at least another project in the past, so we can't pretend the risk is low or does not exist.

So, I'd very much like to know our two project admins' opinion on the matter: @xfurry and @cyberium.

**Comment by killerwife** [comment]
@cyberium already expressed his concerns, I talked to several people about it. Our a lot more popular competitor TC has them running since 2014 without issue. However, I will not be putting a nail in peoples eyes and will move it to a discord posting if I am told to do so. My main goals here are mostly of a long terms nature, and if anyone wishes to know them just ask on discord and I will explain the reasons why I even bothered with the bounty to begin with.

**Comment by killerwife** [comment]
@cala After a while I have a nice update. With some cooperation with @insunaa and @MantisLord we stomped the first 3 phases of the War Effort. What remains is scripting the 10 hour war event. I only did two parts, I added spawns for all the pylons in the correct zones and I scripted it so that the big boss quests are available after killing them.

What is missing is spawning of correct npcs at each pylon, their drops (which are significantly better than normal spawns) and the happenings in silithus, that being all the bosses, elites, and the starting RP. Most of this can be found already in vmangos. I myself will not be touching the 10 hour war for sanity reasons.
